src/interface/ui/accountactivity/BlockchainAccountActivity.cpp

```diff
@@ -424,23 +427,33 @@ auto BlockchainAccountActivity::process_txid(const Message& in) noexcept -> void
     wait_for_startup();
     const auto body = in.Body();
 
-    OT_ASSERT(2 < body.size());
+    OT_ASSERT(3 < body.size());
 
-    const auto txid = Widget::api_.Factory().Data(body.at(1));
+    const auto& api = Widget::api_;
+    const auto txid = api.Factory().Data(body.at(1));
     const auto chain = body.at(2).as<blockchain::Type>();
 
     if (chain != chain_) { return; }
 
-    process_txid(txid);
+    const auto proto = proto::Factory<proto::BlockchainTransaction>(body.at(3));
+    process_txid(txid, factory::BitcoinTransaction(api, proto));
 }
 
 auto BlockchainAccountActivity::process_txid(const Data& txid) noexcept
     -> std::optional<AccountActivityRowID>
+{
+    return process_txid(
+        txid, Widget::api_.Crypto().Blockchain().LoadTransactionBitcoin(txid));
+}
+
+auto BlockchainAccountActivity::process_txid(
+    const Data& txid,
+    std::unique_ptr<const blockchain::bitcoin::block::Transaction> pTX) noexcept
```

Comment:
For clarity, should that unique_ptr be passed as rvalue ref since it is uncopyable anyway? Same in similar cases below...

Reply:
MT-84

Reply:
If I  get assigned to MT=84 in a month, I won't have the foggiest as to what to do.